flang/lib/Lower/OpenMP.cpp

```diff
@@ -2248,11 +2248,15 @@ static void createBodyOfOp(
     if (clauses)
       ClauseProcessor(converter, *clauses).processCopyin();
   }
+
+  if (genNested)
+    genNestedEvaluations(converter, eval);
 }
 
 static void genBodyOfTargetDataOp(
     Fortran::lower::AbstractConverter &converter,
-    Fortran::lower::pft::Evaluation &eval, mlir::omp::DataOp &dataOp,
+    Fortran::lower::pft::Evaluation &eval, bool genNested,
```

Comment:
Every call seems to ultimately set `genNested=true`, is it needed here?

Reply:
I made it so that the `genNested` argument is set in the genOMP functions and then passed on to all nested calls.  This is to make the calls have a more consistent treatment.

If we even do the breakup where each function creates only one op, then this argument will always be `true`, and can be eliminated.